crates/turbopack-node/src/pool.rs

```diff
@@ -311,35 +294,52 @@ pub struct NodeJsOperation {
     #[allow(dead_code)]
     permit: OwnedSemaphorePermit,
     processes: Arc<Mutex<Vec<NodeJsPoolProcess>>>,
+    allow_process_reuse: bool,
 }
 
 impl NodeJsOperation {
-    fn process_mut(&mut self) -> Result<&mut RunningNodeJsPoolProcess> {
-        self.process
+    async fn with_process<'a, F: Future<Output = Result<T>> + Send + 'a, T>(
```

Comment:
This API could also use a `ProcessGuard<'a>`, right?

Reply:
I tried, but it can't capture the `Result` Err in a great way.

```
let guard = op.process();

let result = some_stuff_that_might_fail(*guard).await?;

// You would need to actively flag the guard as successful
guard.ok();

// So dropping guard knows that it should not set allow_process_reuse to false
```

Compare to:

```
let result = op
  .with_process(
      |process| async move { some_stuff_that_might_fail(process).await },
  )
  .await?;
```